### DISCUSSION TITLE
Comments on line start are not highlighted correctly

### DISCUSSION THREAD

**Discussion by SaumitraLohokare** [helix-editor/helix#12128] [Troubleshooting]
### Summary

Comments at the beginning of a line aren't highlighted properly if there's another comment on the previous line. I've noticed this issue in Rust, but I haven't checked if it happens in other languages.

![image](https://github.com/user-attachments/assets/ba7abb23-aa48-4df5-b6ec-f6997cd03ed9)

### Reproduction Steps

 

I tried this:

1. Add single line comments at the very start of the lines for multiple lines.

I expected this to happen:

All the comments should get highlighted correctly

Instead, this happened:

The first character of all comments (except the first one) is not highlighted correctly.

### Platform

Windows

### Terminal Emulator

Windows Terminal App

### Installation Method

release page

### Helix Version

helix 24.7 (079f5442)

**Comment by TornaxO7** [comment]
1. Have you tried it with the master branch?
2. Can you please provide a small example file to test it out locally? Because I can't reproduce this behaviour.

**Comment by SaumitraLohokare** [comment]
1. I tried building master branch and but I was unable to get highlighting to work there. Once I fix my installation mistakes I'll get back to you.

2. I added the exact file that was causing this issue (`test.rs`). I am also attaching a video that demonstrates my steps.

https://github.com/user-attachments/assets/947eac96-24ca-4e0f-bb1b-22651176a4d6

**NOTE**: I renamed this file to `.txt` because github does not allow uploading `.rs`. Please change the extension after downloading.
[test.txt](https://github.com/user-attachments/files/17906533/test.txt)

**Comment by ondro-xy** [comment]
This looks like #10992 Incorrect syntax highlighting with CRLF line endings

**Comment by TornaxO7** [comment]
> 2. I added the exact file that was causing this issue (test.rs). I am also attaching a video that demonstrates my steps.

I can't reproduce it on master.

**Comment by RoloEdits** [comment]
I also can't reproduce on master.

**Comment by kirawi** [comment]
Are you using a custom `languages.toml`?
